src/Microsoft.Diagnostics.ExtensionCommands/HeapWithFilters.cs

```diff
@@ -228,6 +233,11 @@ public IEnumerable<ClrObject> EnumerateFilteredObjects(CancellationToken cancell
                     objs = segment.EnumerateObjects(carefully: true);
                 }
 
+                if (Generation is Generation generation)
+                {
+                    objs = objs.Where(obj => segment.GetGeneration(obj.Address) == generation);
```

Comment:
nitpick:  You can use `obj` here and drop `.Address`, which matches the style of the rest of SOS.  (ClrObject has an implicit operator to convert to ulong.)